### PR TITLE
Retargeting: refactor redirect user composition

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Add `RedirectUser` component which can be conditionally rendered

## Justify why they are needed

- I want to handle not having shop session ID in a single place

- We could potentially do this also by throwing an error but that will leave some race conditions unaccounted for...

- Let me know what you think of this approach!

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3085** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3085" target="_blank">Graphite</a> 
    * **PR #3086** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3086" target="_blank">Graphite</a> 
      * **PR #3087** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3087" target="_blank">Graphite</a> 
        * **PR #3088** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3088" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3088?utm_source=stack-comment).

File: apps/store/src/pages/session/resume.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ const Page = (props: Props) => {
         <meta name="robots" content="noindex,follow" />
       </Head>
       <RetargetingPage {...props} />
+      {redirectUserProps && <RedirectUser {...redirectUserProps} />}
```
Comment (alebedev) [inline apps/store/src/pages/session/resume.tsx:28]: Reminds me of react-router where everything was a component.  Not sure this is most readable way to organize routing.  But it's nice way to sync with React state or redirect conditionally

Do you think this is more clear way to do it?

Can we concentrate "should I redirect" and "do a redirect" in single place?

Comment (robinandeer) [reply apps/store/src/pages/session/resume.tsx:28]: Can we merge like this and later refactor? @alebedev

Comment (alebedev) [reply apps/store/src/pages/session/resume.tsx:28]: Sure, let's make it better later

**Review by alebedev — APPROVED** [review]

**Pull request merged**